_posts/2019-02-27-Extending-Sesam-integration-platform-with-custom-data-sink-service.md

```diff
@@ -0,0 +1,182 @@
+    logging.info("starting service on port %d", port)
+    serve(APP, host='0.0.0.0', port=port)
+```
+
+    INFO:root:starting service on port 5000
+    
+
+    Serving on http://LAPTOP-DL951UEU:5000
+    
+
+This service has one endpoint that serves POST requests and takes queue name as path parameter. Payload must be an array of JSON objects. You may also assign and use environmental variable _PAYLOAD\_KEY_ if you don't need to send a whole Sesam entity with its metadata but only a part of it.
+
+Next thing we need to do complete our service is to create Docker image of it and push it to a Docker repository where Sesam will be able to pull it. I'm going to use public Dockerhub repository, but images may be placed in any docker repository including private repositories.  
+
+Add Dockerfile to your project and build & push image.
+
+```
+FROM python:3-slim
+
+COPY ./* /service/
+WORKDIR /service
+
+RUN pip install -r requirements.txt
+
+EXPOSE 5000/tcp
+
+ENTRYPOINT ["python3"]
+CMD ["service.py"]
+```
+
+Now it's time to use our newly created service to send some data from Sesam to Azure Service Bus.
+
+Login in into Sesam portal and join free of charge "Open Sesam" subscription (this subscription is for open data so don't use it in production or with sensitive data).
+Press "Systems" and create new system using following system definition.
+```json
+{
+  "_id": "azure-sb-sink",
+  "type": "system:microservice",
+  "docker": {
+    "environment": {
+      "SERVICE_NAMESPACE": "$SECRET(service_namespace)",
+      "SERVICE_SAS_TOKEN_NAME": "$SECRET(service_sas_token_name)",
+      "SERVICE_SAS_TOKEN_VALUE": "$SECRET(service_sas_token_value)"
+    },
+    "image": "<location of your image>",
+    "port": 5000
+  },
+  "verify_ssl": true
+}
+
+```
+Press "Save" - you system is stored. Then open "Secrets" tab and add following items to Sesam vault:
+* service_namespace - with your Azure Service Bus namespace
+* service_sas_token_name - with SAS token name
+* service_sas_token_value - with content of SAS token
+
+Now your system will be able to use them without exposing as plain text to everyone including yourself.
+
+We have now a data sink system and next step is to send some data to it. Open "Pipes" menu and create new pipe using following pipe definition: 
+```json
+{
+  "_id": "azure-sb-sink-pipe",
+  "type": "pipe",
+  "source": {
+    "type": "embedded",
+    "entities": [{
+      "_id": "1",
+      "value": "foo"
+    }, {
+      "_id": "2",
+      "value": "bar"
+    }]
+  },
+  "sink": {
+    "type": "json",
+    "system": "azure-sb-sink",
+    "url": "my_queue"
+  },
+  "pump": {
+    "schedule_interval": 60
+  }
+}
+```
+We will use embedded data set with 2 simple entities.  
+That's all! Now this pipe will send its embedded data to Azure Service Bus every 60 seconds.  
+In this post we described how to extend functionality of Sesam integration platform by adding a custom data sink connector with about 50 lines of code.  
```

Comment:
I suggest a bit longer summary here- with a short walk through for example..This is to be able to address the ones who skips the implementation details.

Reply:
@egemenyavuz added som lines of text